### PR TITLE
NodeFeatureRules add new provided API

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/openshift/origin-cluster-nfd-operator
+  newName: quay.io/eduardoarango/cluster-nfd-operator
```
Comment (kpouget) [inline config/manager/kustomization.yaml:15]: @ArangoGutierrez is that on purpose?

Comment (ArangoGutierrez) [reply config/manager/kustomization.yaml:15]: off course..... 😎

**Comment by kpouget** [comment]
thanks @ArangoGutierrez, the content looks consistent (bear with me, I skipped the `vendor/` directory 🙃)

/lgtm
/hold unhold when you're ready for merging

**Comment by lenahorsley** [comment]
Ran the following tests:
1. Created and deployed a NFD bundle from https://github.com/ArangoGutierrez/cluster-nfd-operator/tree/devel/4.10
2. Checked the worker node labels (three worker nodes were labeled).
3. Added node/machineset and new node was labeled. New NFD pod in openshift-nfd namespace.
4. Removed newly added worker node. Remaining worker nodes were labeled.
5. Log creation and stability
6. Pods in etcd namespace did not restart.
All tests passed.

**Pull request merged**